### PR TITLE
Eslint: upgrade to version 8.0.0

### PR DESCRIPTION
**Pull request by mrtnzlml**
See:

- https://eslint.org/blog/2021/10/eslint-v8.0.0-released
- https://eslint.org/docs/user-guide/migrating-to-8.0.0

Known unresolved issues (with low impact )that we decided to accept:

- https://github.com/mysticatea/eslint-plugin-node/issues/301

Known issues that need to be fixed first:

- [x] wait for https://github.com/import-js/eslint-plugin-import/pull/2191 to be released and merged here
- [x] wait for https://github.com/gajus/eslint-plugin-flowtype/pull/496 to be released and merged here
- [x] wait for https://github.com/testing-library/eslint-plugin-testing-library/issues/462 to be released and merged here
- [ ] ~wait for https://github.com/mysticatea/eslint-plugin-node/issues/294 to be released and merged here~ replaced (probably temporarily) with `eslint-plugin-n` which supports Eslint 8
- [x] wait for https://github.com/facebook/react/pull/22248 to be released and merged here
- [x] release minor/patch version of Adeira Eslint Config before merging this breaking change
- [x] switch `NEXT_VERSION_ERROR` to `ERROR` (https://github.com/adeira/universe/pull/3625)

### CODE REVIEW

**Comment by maicol07** [comment]
@mrtnzlml it seems there is a new eslint plugin for node, as the original one seems unmaintained: https://www.npmjs.com/package/eslint-plugin-n

**Comment by mrtnzlml** [comment]
> @mrtnzlml it seems there is a new eslint plugin for node, as the original one seems unmaintained: https://www.npmjs.com/package/eslint-plugin-n

https://github.com/adeira/universe/pull/3624

File: yarn.lock [hunk 1]
```diff
@@ -9062,60 +9073,58 @@ eslint-visitor-keys@^3.0.0:
   resolved "https://registry.yarnpkg.com/eslint-visitor-keys/-/eslint-visitor-keys-3.0.0.tgz#e32e99c6cdc2eb063f204eda5db67bfe58bb4186"
   integrity sha512-mJOZa35trBTb3IyRmo8xmKBZlxf+N7OnUl4+ZhJHs/r+0770Wh/LEACE2pqMGMe27G/4y8P2bYGk4J70IC5k1Q==
 
-eslint@^7.32.0:
```
Comment (mrtnzlml) [inline yarn.lock:9065]: Eslint 7 was correctly removed.

File: yarn.lock [hunk 2]
```diff
@@ -9062,60 +9073,58 @@ eslint-visitor-keys@^3.0.0:
...
-  version "7.32.0"
-  resolved "https://registry.yarnpkg.com/eslint/-/eslint-7.32.0.tgz#c6d328a14be3fb08c8d1d21e12c02fdb7a2a812d"
-  integrity sha512-VHZ8gX+EDfz+97jGcgyGCyRia/dPOd6Xh9yPv8Bl1+SoaIwD+a/vlrOmGRUyOYu7MwUhc7CxqeaDZU13S4+EpA==
+eslint@^8.0.0:
```
Comment (mrtnzlml) [inline yarn.lock:9076]: Eslint 8 was correctly added.

**Review by tbergquist-godaddy — APPROVED** [review]

**Comment by mrtnzlml** [comment]
Hola @Brianzchen 👋 Would you happen to know are the tests failing because of `eslint-plugin-ft-flow` on Eslint 8? 🤔 Is `eslint-plugin-ft-flow` tested against Eslint 8? Does it support this latest Eslint version?

**Comment by Brianzchen** [comment]
So I raised an issue about this [here](https://github.com/flow-typed/eslint-plugin-ft-flow/issues/9). I guess I understand what this dodgy code does now. My projects are on eslint v8 but I suppose it never impacted me because I never used this rules.

This is an easy fix let me raise a PR for you shortly

File: src/eslint-config-adeira/src/presets/flowtype.js [hunk 3]
```diff
@@ -27,7 +29,7 @@ module.exports = ({
     'ft-flow/no-duplicate-type-union-intersection-members': ERROR,
     'ft-flow/no-existential-type': ERROR, // https://github.com/facebook/flow/issues/6308
     'ft-flow/no-flow-fix-me-comments': OFF,
-    'ft-flow/no-flow-fix-me-in-strict-files': OFF, // TODO: enable (after fixing our codebase)
+    'ft-flow/no-flow-suppressions-in-strict-files': OFF, // TODO: enable (after fixing our codebase)
```
Comment (mrtnzlml) [inline src/eslint-config-adeira/src/presets/flowtype.js:32]: BTW @Brianzchen - seems like you are about to release this breaking change in a patch version. 😬

File: src/eslint-config-adeira/package.json [hunk 4]
```diff
@@ -21,7 +21,7 @@
     "eslint-plugin-adeira": "^0.15.0",
     "eslint-plugin-eslint-comments": "^3.2.0",
     "eslint-plugin-fb-flow": "^0.0.4",
-    "eslint-plugin-ft-flow": "^1.1.0",
+    "eslint-plugin-ft-flow": "^1.1.1-alpha-0",
```
Comment (mrtnzlml) [inline src/eslint-config-adeira/package.json:24]: TODO: wait for a proper release

File: src/eslint-config-adeira/src/presets/flowtype.js (hunk 3 above)
Comment (Brianzchen) [reply src/eslint-config-adeira/src/presets/flowtype.js:32]: Thanks forgot about this one. Yes I'll have to release this under v2.0.0 then if that works for you

File: src/eslint-config-adeira/package.json (hunk 4 above)
Comment (Brianzchen) [reply src/eslint-config-adeira/package.json:24]: Released to [v2.0.0](https://github.com/flow-typed/eslint-plugin-ft-flow/releases/tag/v2.0.0)

**Comment by maicol07** [comment]
@mrtnzlml I think this can be merged now that all the plugins have been updated!

**Pull request merged**